### ISSUE TITLE
Custom render control leaves "spots" in background

### ISSUE THREAD

**Issue by llfab** [AvaloniaUI/Avalonia#9659]
**Describe the bug**
I created a custom render templated control which represents the Windows 10 busy indicator with circulating dots. This uses ```context.DrawEllipse```. After each circle of the dots a spot in the color of the dots is left in the background. This spot becomes more and more visible after each circle iteration. Hence, it seems that the background is not correctly cleared. This is visible differently on different machines. Sometimes its in the top row, sometimes its left, sometimes it is both. NOTE: the problem does not appear in the designer view within Visual Studio

**To Reproduce**
Use the busy indicator and set to active.

**Expected behavior**
The "spots" should not appear.

**Screenshots**
I have created screenshots during active animation. I have set the background to darg gray to visualize that the spot happens within the controls view port.
![image](https://user-images.githubusercontent.com/31031996/206695093-13b73e26-9897-46fe-8dfa-a1dfb28fcfb7.png)

Note: the spots appear outside the dot drawing area which is set to view port - 1.
![image](https://user-images.githubusercontent.com/31031996/206695364-5724b2c4-1c3b-4ec3-8e55-6b3125dc6ed4.png)

**Desktop (please complete the following information):**

- OS: Windows 10
- Version: 11.0.0-preview4

**Additional context**
This is the custom render method:
```csharp
        public override void Render(DrawingContext context)
        {
            if (!IsActive || DotCount == 0) { return; }

            double size = Math.Min(Bounds.Width, Bounds.Height);
            double dotRadius = 0.05 * size;
            Circle2D circle = new Circle2D(0.5 * Bounds.Width, 0.5 * Bounds.Height, 0.5 * size - dotRadius - 1);

            Interval range = new Interval(Phase, Phase + 0.06 * (DotCount - 1));

            foreach (double t in range.Discretize(DotCount))
            {
                Point2D p = circle.GetPointForAngle(GetAngleAt(t));
                context.DrawEllipse(Brush, null, new Point(p.X, p.Y), dotRadius, dotRadius);
            }
        }
```

You can find the full code of the busy indicator here: https://github.com/llfab/Samples/tree/main/BusyIndicator

**Comment by llfab** [comment]
Please find these sample videos:

https://user-images.githubusercontent.com/31031996/206697764-176507c7-3a7e-45ad-8825-b18ceaf15425.mp4

And another one with a different setup:

https://user-images.githubusercontent.com/31031996/206697871-5779a2d3-7e6f-4cb3-993e-5710e7a55799.mp4

**Comment by maxkatz6** [comment]
Can you check with latest nighly?

**Comment by llfab** [comment]
where again to I find the nightly NuGet package names?

**Comment by kekekeks** [comment]
You need to provide accurate render operation bounds. Otherwise the renderer wont invalidate the entire area

**Comment by llfab** [comment]
Kekekeks: "You need to provide accurate render operation bounds. Otherwise the renderer wont invalidate the entire area".

How would I do that?

**Comment by llfab** [comment]
So, this works:
```csharp
        public override void Render(DrawingContext context)
        {
            if (!IsActive || DotCount == 0) { return; }

            double size = Math.Min(Bounds.Width, Bounds.Height);
            double dotRadius = 0.05 * size;
            Circle2D circle = new Circle2D(0.5 * Bounds.Width, 0.5 * Bounds.Height, 0.5 * size - dotRadius - 1);

            Interval range = new Interval(Phase, Phase + 0.06 * (DotCount - 1));

            // draw transparent black in the entire view port
            context.FillRectangle(BackgroundBrush, Bounds);

            foreach (double t in range.Discretize(DotCount))
            {
                Point2D p = circle.GetPointForAngle(GetAngleAt(t));
                context.DrawEllipse(ForegroundBrush, null, new Point(p.X, p.Y), dotRadius, dotRadius);
            }
        }
```

Is it meant like that?

**Comment by kekekeks** [comment]
Its' not meant to be like this and is likely to be some bug in ellipse bounds calculations, but filling the entire background is a good workaround.

**Comment by llfab** [comment]
Double checked with latest nightly. The problem also exists there (which was expected, I guess) and still needs the fill background workaround.

**Comment by timunie** [comment]
I had a similar issue in the past: https://github.com/AvaloniaUI/Avalonia/pull/7778 . Maybe your fix will be similar to my fix. You can always use DevTools [F12] in debug mode to render dirty rects. these indicate which area will be drawn again.

**Comment by llfab** [comment]
I checked with F12. The "curtain" window that contains this busy indicator does not show any dirty rect during busy indicator animation.

**Comment by timunie** [comment]
@llfab I wanted to try the sample you attached. The following code is missing: 

- Circle2D
- Interval
- Point2D
- MathUtils

Please provide this code / classes, so we can test the same environment as you do. Thx.

**Comment by llfab** [comment]
This should now be self-contained

Note: this seems to happen not an all machines and not in the same fashion

**Comment by llfab** [comment]
https://github.com/llfab/Samples/tree/main/BusyIndicator

**Comment by timunie** [comment]
will you be able to test my idea on a machine where you know it fails? I pushed an update to this PR https://github.com/AvaloniaUI/Avalonia/pull/9686. Probably you can check out the branch and test it there.

**Comment by llfab** [comment]
I will not be able to consume from source but if the PR would create a nuget 11.999-beta... then I can certainly try that

**Comment by timunie** [comment]
Sure, here it goes: https://github.com/AvaloniaUI/Avalonia/pull/9686#issuecomment-1348937402 

But you will need to add the all nuget feed (feed url in the comment). Please note that my fix is judged as a workaround and may not be merged as is. But it'd be great to verify that we're on the right track.

**Comment by llfab** [comment]
I tested it on my machine where it always happens. It seems the PR does not change anything. See images:

Preview4:
![Preview4](https://user-images.githubusercontent.com/31031996/207551446-bb779f31-2cb2-46f8-84bf-33cfe132dc83.png)

PR (11.0.999-cibuild0027551-beta):
![11 0 999-cibuild0027551-beta](https://user-images.githubusercontent.com/31031996/207551483-e3b41af3-282f-44d0-a16b-1aa5724a2124.png)

**Comment by llfab** [comment]
Thanks for taking this on anyways!

**Comment by timunie** [comment]
Should be fixed in upcoming v11.1 acc. To customer feedback
